MoreLinq.Test/TransposeTest.cs

```diff
@@ -0,0 +1,232 @@
+#region License and Terms
+// MoreLINQ - Extensions to LINQ to Objects
+// Copyright (c) 2018 Leandro F. Vieira (leandromoh). All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+namespace MoreLinq.Test
+{
+    using System;
+    using System.Collections.Generic;
+    using NUnit.Framework;
+
+    [TestFixture]
+    public class TransposeTest
+    {
+        [Test]
+        public void TransposeIsLazy()
+        {
+            new BreakingSequence<BreakingSequence<int>>().Transpose();
+        }
+
+        [Test]
+        public void TransposeWithNullInnerSequence()
+        {
+            var matrix = new[]
+            {
+                new [] { 10, 11 },
+                new int[0],
+                new [] { 30, 31, 32 },
+                null
+            };
+
+            Assert.Throws<NullReferenceException>(() =>
+                matrix.Transpose().FirstOrDefault());
+        }
+
+        [Test]
+        public void TransposeWithEqualsLengthEnumerables()
```

Comment:
Rename to either `TransposeWithRegularSequences` or `TransposeWithSequencesOfSameLength`.